services/net/contentmigration/migrators/ImageMigrator.cs

```diff
@@ -0,0 +1,255 @@
+    /// Sends message to kafka if content has been added or updated.
+    /// Informs API of content reference status.
+    /// </summary>
+    /// <param name="manager"></param>
+    /// <param name="lookups"></param>
+    /// <param name="ingests"></param>
+    /// <param name="newsItem"></param>
+    /// <returns></returns>
+    public async Task MigrateNewsItemAsync(
+        IIngestServiceActionManager manager,
+        LookupModel? lookups,
+        IEnumerable<IngestModel> ingests,
+        NewsItem newsItem)
+    {
+        if (lookups == null) {
+            this.Logger.LogError("Lookups cannot be null");
+            throw new ArgumentNullException(nameof(lookups));
+        }
+
+        try
+        {
+            SourceModel? source = GetSourceMapping(lookups.Sources, newsItem.Source);
+            if (source == null) {
+                // if we don't have a custom mapping, fallback to our Ingest 'DEFAULT' mapping
+                // TODO: KGM - When this happens, store the TNO 1.0 "Source" in "Other Source" field
+                source = GetSourceMapping(lookups.Sources, manager.Ingest.Source!.Code);
+                if (source == null) {
+                    this.Logger.LogWarning("Couldn't map to Source for NewsItem with source '{sourceName}'", newsItem.Source);
+                    return;
+                }
+            }
+            ProductModel? product = GetProductMapping(lookups.Products, newsItem);
+            if (product == null) {
+                this.Logger.LogWarning("Couldn't map to Product for NewsItem with type '{sourceName}'", newsItem.Type);
+                return;
+            }
+
+            // KGM: does uid exist for all newsItem types in old system and does this correlate to new system?
+            // KGM: for soft-launch - how do we avoid duplication if MMIA is ingest from source *AND* historic?
+            var uid = newsItem.WebPath ?? $"{newsItem.RSN}";
+            var reference = await this.FindContentReferenceAsync(source?.Code, uid);
+            if (reference == null)
+            {
+                reference = await this.Api.AddContentReferenceAsync(CreateContentReference(source!, manager.Ingest.Topic, newsItem, uid));
+                Logger.LogInformation("Migrating content {RSN}:{Title}", newsItem.RSN, newsItem.Title);
+            }
+            else if (reference.Status == (int)WorkflowStatus.InProgress && reference.UpdatedOn?.AddMinutes(5) < DateTime.UtcNow)
+            {
+                // If another process has it in progress only attempt to do an Migration if it's
+                // more than an 5 minutes old. Assumption is that it is stuck.
+                reference = await UpdateContentReferenceAsync(reference, WorkflowStatus.InProgress);
+                Logger.LogInformation("Updating migrated content {RSN}:{Title}", newsItem.RSN, newsItem.Title);
+            }
+
+            reference = await FindContentReferenceAsync(reference?.Source, reference?.Uid);
```

Comment:
Not sure we need to make a second request for content reference here.  I wonder if this was introduced by Cello when debugging concurrency issues.